lib/hanami/cli/commands/monolith/server.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require "rack"
+require_relative "../application"
+require_relative "../../server"
+
+module Hanami
+  module CLI
+    module Commands
+      module Monolith
+        # Launch Hanami web server.
+        #
+        # The server is just a thin wrapper on top of Rack::Server. The options that it
+        # accepts fall into two different categories:
+        #
+        # - When not explicitly set, port and host are not passed to the rack
+        # server instance. This way, they can be configured through the
+        # configured rack handler (e.g., the puma configuration file).
+        #
+        # - All others are always given by the Hanami command.
+        #
+        # Run `bundle exec hanami server -h` to see all the supported options.
+        class Server < Application
+          desc "Start Hanami server"
+
+          option :host, default: nil, required: false, desc: "The host address to bind to (falls back to the rack handler)"
+          option :port, default: nil, required: false, desc: "The port to run the server on (falls back to the rack handler)"
+          option :config, default: 'config.ru', required: false, desc: "Rack configuration file"
+          option :daemonize, default: false, required: false, desc: "Daemonize the server"
+          option :debug, default: false, required: false, desc: "Turn on/off debug output"
+          option :pid, default: nil, required: false, desc: "Path to write a pid file after daemonize"
+          option :warn, default: false, required: false, desc: "Turn on/off warnings"
+
+          # @api private
+          def call(**options)
+            Hanami::CLI::Server.new.call(**options)
```

Comment:
@waiting-for-dev Just a stylistic feedback regarding showcasing `dry-cli`.

```ruby
private attr_reader :server

def initialize(server: Hanami::CLI::Server.new)
  @server = server
end

def call(...)
  server.call(...)
end
```

Reply:
Thanks for the suggestion. Done!